validator/rpc/server.go

```diff
@@ -170,50 +172,50 @@ func (s *Server) InitializeRoutes() error {
 		return errors.New("no router found on server")
 	}
 	// Adding Auth Interceptor for the routes below
-	s.router.Use(s.AuthTokenHandler)
+	s.server.RegisterMiddlewares(s.AuthTokenHandler)
```

Comment:
this isn't getting properly registered, you'd have to put it onto the middleware chain

Reply:
doing something like `httprest.WithMiddlewares(append(cfg.Middlewares, server.AuthTokenHandler)),` should work